lib/internal/bootstrap/node.js

```diff
@@ -23,11 +23,91 @@
                                 _memoryUsage, _rawDebug,
                                 _umask, _initgroups, _setegid, _seteuid,
                                 _setgid, _setuid, _setgroups,
-                                _shouldAbortOnUncaughtToggle },
+                                _shouldAbortOnUncaughtToggle,
+                                _setupPrepareStackTrace },
                               { internalBinding, NativeModule },
                               triggerFatalException) {
   const exceptionHandlerState = { captureFn: null };
   const isMainThread = internalBinding('worker').threadId === 0;
+  const { safeToString } = internalBinding('util');
+
+  let wrapper;
+  const ErrorConstructor = Error;
+  _setupPrepareStackTrace((globalThis, error, trace, message) => {
```

Comment:
Can you move this into per_context.js, or just somewhere earlier? (but it looks odd in loaders.js) I think the only thing it needs from this closure is `NativeModule.require('module').wrapper[0]`, but it should be easy to put that wrapper string in C++, and pass it onto this callback as well as `cjs/loader.js` through an internalBinding. That enables us to pass wrappers for bootstrappers and native modules from C++ (if we switch on the origin name then we'll know which wrapper to pass).

(On a side note: why is the string manipulation not written in C++? we don't even need errors::CreateMessage if we format this in C++)

Reply:
> Can you move this into per_context.js, or just somewhere earlier?

it actually has to live in the "main" context to simulate the behaviour with `Error.prepareStackTrace` (notice how i check `ErrorConstructor`, regardless of which context its actually called from). Changing that would have noticeable effects in userland, so i was planning to make it a separate pr. At that point, i could move it into per_context.js.

> move string manip into c++

i originally had it in c++, but as the logic got more complex, it kinda naturally fell out to jsland. when i do the aforementioned refactoring i would like to look into cleaning up the decoration, which would probably simplify the path such that it would be nice to do in c++.

this pr is just focused on migrating with minimal behavioural changes beyond what i mention in the original message (double decoration, etc).